libs/core/langchain_core/runnables/base.py

```diff
@@ -2882,8 +2882,71 @@ async def input_aiter() -> AsyncIterator[Input]:
 
 
 class RunnableGenerator(Runnable[Input, Output]):
-    """
-    A runnable that runs a generator function.
+    """A runnable that runs a generator function.
```

Comment:
this is great overall! one detail, we actually support two types of generator functions, might be good to mention that here, see an explanation below

there are two types of generator functions (ie functions that generate a stream of output, as opposed to a function that returns output all at once a single time) that can be useful in building your application:
- "streaming" functions, ie. with signature `A -> Iterator[B]`, they receive the output of the previous step all at once, after the previous step finishes, and can start emitting their own incremental output after that. This is useful when you need the entire output of the previous steps before being able to start doing anything useful in this step, eg. most LLMs need the entire prompt available before starting to generate the completion, so would be easier to implement as one of these. These can be used by wrapping them in RunnableLambda.
- "transformer" functions, ie. with signature `Iterator[A] -> Iterator[B]`, they receive a stream of the output of the previous step while it's being produced, giving them the opportunity to start emitting output chunks as soon as the previous step emits the first chunk. This is mainly useful for output parser-like functions, which are expected to be used after a previous step that streams output, like an LLM, and are expected to parse it incrementally as soon as they can. These can be used by wrapping them in RunnableGenerator.

Reply:
Thanks for this! I added some text making this distinction near the top of the docstring. I also added an example at the bottom capturing the behavior for the first type of function you described. Let me know your thoughts (e.g., if it is too contrived), or if I misunderstood anything. Thanks!

Reply:
Cool, looks great!